src/cljam/util.clj

```diff
@@ -2,17 +2,45 @@
   "General utilities."
   (:require [clojure.java.io :refer [file] :as cio])
   (:import [java.net MalformedURLException URL]
+           [java.nio.file Files FileVisitor FileVisitResult]
+           [java.nio.file.attribute FileAttribute]
            [org.apache.commons.compress.compressors
             CompressorStreamFactory CompressorException]))
 
 ;; Disk cache
 ;; ----------
 
-(def temp-dir (let [system-tmp-dir-path (or (System/getenv "TMPDIR")
-                                            (System/getProperty "java.io.tmpdir"))
-                    dir-path (.getPath (file system-tmp-dir-path "cljam"))]
-                (.mkdirs (file dir-path))
-                dir-path))
+(defmacro with-temp-dir
```

Comment:
The expanded form of this macro looks somewhat large. More importantly, it contains a `reify` form in it, which would redundantly generate a dedicated anonymous class for each macro callsite.

How about breaking the implementation down into some functions like `create-temp-dir` and `delete-temp-dir`, and using them in the expanded form?